### PR TITLE
(refactor) Refactored patient list to display appointment lists, upda…

### PR DESCRIPTION
**Pull request by CynthiaKamau**
…ted missed appointment table rows and added mock data

## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
-Removed patient lists and added appointment list with appointment types tabs
-Refactored appointment table and removed unused code
-Formatted appointment resource and added service types, appointments and appointment fetch response

## Screenshots
appointments

*None.*

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

File: packages/esm-appointments-app/src/tabs/appointments-table.component.tsx [hunk 1]
```diff
@@ -23,36 +19,16 @@ import {
...
-  useVisitQueueEntries,
-  useServices,
-  QueueService,
-  QueueStatus,
-  MappedVisitQueueEntry,
-  MappedQueuePriority,
-  useAppointmentEntries,
-} from './appointments-table.resource';
+import CurrencyPound16 from '@carbon/icons-react/es/currency--pound/16';
+import Omega16 from '@carbon/icons-react/es/omega/16';
```
Comment (makombe) [inline packages/esm-appointments-app/src/tabs/appointments-table.component.tsx:26]: Are these the appropriate icons per the design or these are place holders for the time being?

File: packages/esm-appointments-app/__mocks__/appointments.mock.ts [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+        name: 'Outpatient',
+        description: null,
+        speciality: {},
+        startTime: '',
+        endTime: '',
+        maxAppointmentsLimit: null,
+        durationMins: null,
+        location: {},
+        uuid: 'e2ec9cf0-ec38-4d2b-af6c-59c82fa30b90',
+        color: '#006400',
```
Comment (makombe) [inline packages/esm-appointments-app/__mocks__/appointments.mock.ts:202]: I think for the colors we need to move this to the color style guide. I doubt if we will need to return the color through api endpoints.

File: packages/esm-appointments-app/src/tabs/appointments-table.component.tsx (hunk 1 above)
Comment (CynthiaKamau) [reply packages/esm-appointments-app/src/tabs/appointments-table.component.tsx:26]: These are placeholders

File: packages/esm-appointments-app/src/tabs/appointments-table.component.tsx [hunk 3]
```diff
@@ -361,7 +234,7 @@ function AppointmentsTable() {
         <Tile className={styles.tile}>
           <p className={styles.content}>{t('noAppointmentsToDisplay', 'No appointments to display')}</p>
           <Button kind="ghost" size="small" renderIcon={Add16} onClick={() => setShowOverlay(true)}>
-            {t('addNewAppointment', 'Add New apppointment')}
+            {t('addNewAppointment', 'Add new apppointment')}
```
Comment (makombe) [inline packages/esm-appointments-app/src/tabs/appointments-table.component.tsx:237]:
```suggestion
            {t('addNewAppointment', 'Add new appointment')}
```

File: packages/esm-appointments-app/src/tabs/appointments-table.component.tsx [hunk 4]
```diff
@@ -342,7 +215,7 @@ function AppointmentsTable() {
                     </div>
                     <p className={styles.separator}>{t('or', 'or')}</p>
                     <Button kind="ghost" size="small" renderIcon={Add16} onClick={() => setShowOverlay(true)}>
-                      {t('addNewAppointment', 'Add New apppointment')}
+                      {t('addNewAppointment', 'Add new apppointment')}
```
Comment (makombe) [inline packages/esm-appointments-app/src/tabs/appointments-table.component.tsx:218]:
```suggestion
                      {t('addNewAppointment', 'Add new appointment')}
```

File: packages/esm-appointments-app/src/tabs/appointments-table.resource.ts [hunk 5]
```diff
@@ -90,201 +26,75 @@ interface Encounter {
...
-  // "teleconsultationLink": null,
-  // "recurring": false
+export interface AppointmentsFetchResponse {
+  data: Array<Appointment>;
 }
 
-interface MappedEncounter extends Omit<Encounter, 'encounterType' | 'provider'> {
-  encounterType: string;
-  provider: string;
+export interface Appointment {
```
Comment (donaldkibet) [inline packages/esm-appointments-app/src/tabs/appointments-table.resource.ts:33]: We can move this interface declarations to `types/index.ts` file

File: packages/esm-appointments-app/src/tabs/appointments-table.resource.ts [hunk 6]
```diff
@@ -90,201 +26,75 @@ interface Encounter {
...
+  appointmentNumber: string;
+  comments: string;
+  endDateTime: Date | number;
+  location: OpenmrsResource;
+  patient: fhir.Patient;
+  provider: OpenmrsResource;
+  providers: Array<OpenmrsResource>;
+  recurring: boolean;
+  service: AppointmentService;
+  startDateTime: number | any;
```
Comment (donaldkibet) [inline packages/esm-appointments-app/src/tabs/appointments-table.resource.ts:44]:
```suggestion
  startDateTime: string;
```
Most of the time dates with openmrs rest are returned as string

File: packages/esm-appointments-app/src/tabs/appointments-table.resource.ts [hunk 7]
```diff
@@ -90,201 +26,75 @@ interface Encounter {
...
-    isError: error,
-    isValidating,
-  };
+interface MappedEncounter extends Omit<Encounter, 'encounterType' | 'provider'> {
+  encounterType: string;
+  provider: string;
 }
 
-export function useAppointmentEntries(): UseVisitQueueEntries {
+export function useAppointmentEntries() {
```
Comment (donaldkibet) [inline packages/esm-appointments-app/src/tabs/appointments-table.resource.ts:76]:
```suggestion
export function useAppointments() {
```

File: packages/esm-appointments-app/src/tabs/appointments-table.resource.ts [hunk 8]
```diff
@@ -90,201 +26,75 @@ interface Encounter {
...
-    visitType: visitQueueEntry.visit?.visitType?.display,
-    visitUuid: visitQueueEntry.visit?.uuid,
-  });
-
-  const mappedVisitQueueEntries = data?.data?.results?.map(mapVisitQueueEntryProperties);
+  const mappedAppointmentEntries = mockAppointmentsData.data?.map(mapAppointmentProperties);
 
   return {
-    visitQueueEntries: mappedVisitQueueEntries ? mappedVisitQueueEntries : null,
+    appointmentEntries: mappedAppointmentEntries ? mappedAppointmentEntries : null,
```
Comment (donaldkibet) [inline packages/esm-appointments-app/src/tabs/appointments-table.resource.ts:97]:
```suggestion
    appointments: mappedAppointments ? mappedAppointment : [],
```

**Review by donaldkibet — COMMENTED** [review]
LGTM Thanks @CynthiaKamau just some few suggestions

**Review by donaldkibet — APPROVED** [review]
Thanks @CynthiaKamau

**Comment by gracepotma** [comment]
FYI @samuelmale you may be interested in watching this :)

**Pull request merged**